cmd/agent/main.go

```diff
@@ -300,18 +301,26 @@ func startModelPuller(logger *zap.SugaredLogger) {
 		}
 	}
 
+	var gcs_client *gstorage.Client
+	var err error
+
+	ctx := context.Background()
 	if _, ok := os.LookupEnv(gcscredential.GCSCredentialEnvKey); ok {
 		// GCS relies on environment variable GOOGLE_APPLICATION_CREDENTIALS to point to the service-account-key
 		// If set, it will be automatically be picked up by the client.
 		logger.Info("Initializing gcs client, using existing GOOGLE_APPLICATION_CREDENTIALS variable.")
-		ctx := context.Background()
-		client, err := gstorage.NewClient(ctx)
-		if err != nil {
-			panic(err)
-		}
-		downloader.Providers[storage.GCS] = &storage.GCSProvider{
-			Client: stiface.AdaptClient(client),
-		}
+		gcs_client, err = gstorage.NewClient(ctx)
```

Comment:
I think it is better to initialize the storage client [here](https://github.com/kubeflow/kfserving/blob/master/pkg/agent/downloader.go#L65), when protocol is extracted we know what storage client to initialize, same applies to S3.